core/tee/tee_svc_cryp.c

```diff
@@ -871,6 +898,121 @@ static void op_attr_value_clear(void *attr)
 	*v = 0;
 }
 
+static TEE_Result op_attr_25519_from_user(void *attr, const void *buffer,
+					  size_t size)
+{
+	uint8_t **key = attr;
+
+	if (size != KEY_SIZE_BYTES_25519 || !*key)
+		return TEE_ERROR_SECURITY;
+
+	memcpy(*key, buffer, size);
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result op_attr_25519_to_user(void *attr,
+					struct ts_session *sess __unused,
+					void *buffer, uint64_t *size)
+{
+	TEE_Result res;
+	uint8_t **key = attr;
+	uint64_t s = 0;
+	uint64_t key_size = (uint64_t)KEY_SIZE_BYTES_25519;
+
+	res = copy_from_user(&s, size, sizeof(s));
+	if (res != TEE_SUCCESS)
+		return res;
+
+	res = copy_to_user(size, &key_size, sizeof(key_size));
+	if (res != TEE_SUCCESS)
+		return res;
+	if (!key_size)
```

Comment:
is it possible that `key_size == 0` here? Did you mean `if (!s)`?

Reply:
thanks for pointing out , this check `if (!key_size)` is not needed, will remove!